internal/clients/telegram/client.go

```diff
@@ -0,0 +1,282 @@
+package telegram
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"net/url"
+	"path"
+	"strconv"
+	"sync"
+	"time"
+)
+
+// A client to interact with the telegram API. Start the client and then call Stop to stop it gracefully.
+// The client may take some time to shutdown if it has work to do.
+//
+// # Example
+//
+//	client := telegram.NewClient("api.telegram.org", "TOKEN")
+//	client.Start(10) // 10 threads
+//
+//	if shouldStop { // whatever the trigger for stop is
+//	    client.Stop()
```

Comment:
learn patterns! and look into: https://www.digitalocean.com/community/tutorials/how-to-use-interfaces-in-go

Reply:
this is what i want!